tools/fuzzer/platform/imx8-host.c

```diff
@@ -0,0 +1,515 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+// Copyright 2020 NXP
+//
+// Author: Diana Cretu <diana.cretu@nxp.com>
+
+/* Core IA host MU support for i.MX8 audio DSP. */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <pthread.h>
+#include <string.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include "shim.h"
+#include "mu.h"
+#include <ipc/trace.h>
+#include <ipc/info.h>
+#include "../fuzzer.h"
+#include "../qemu-bridge.h"
+
+#define MBOX_OFFSET		0x144000
+
+#define ADSP_IMX8_DSP_MAILBOX_BASE      0x92C00000
+#define ADSP_IMX8_HOST_IRAM_OFFSET      0x10000
+#define ADSP_IMX8_HOST_IRAM_BASE        0x596f8000
+#define ADSP_IMX8_HOST_DRAM_BASE        0x596e8000
+#define SDRAM0_BASE     0x92400000
+#define SDRAM0_SIZE     0x800000
+#define SDRAM1_BASE     0x92C00000
+#define SDRAM1_SIZE     0x800000
+
+/* Mailbox configuration */
+#define ADSP_SRAM_OUTBOX_BASE       ADSP_IMX8_DSP_MAILBOX_BASE
+#define ADSP_SRAM_OUTBOX_SIZE       0x1000
+#define ADSP_SRAM_OUTBOX_OFFSET     0
+
+#define ADSP_SRAM_INBOX_BASE        (ADSP_SRAM_OUTBOX_BASE \
+					+ ADSP_SRAM_OUTBOX_SIZE)
+#define ADSP_SRAM_INBOX_SIZE        0x1000
+#define ADSP_SRAM_INBOX_OFFSET      ADSP_SRAM_OUTBOX_SIZE
+
+#define ADSP_SRAM_DEBUG_BASE        (ADSP_SRAM_INBOX_BASE \
+				     + ADSP_SRAM_INBOX_SIZE)
+#define ADSP_SRAM_DEBUG_SIZE        0x800
+#define ADSP_SRAM_DEBUG_OFFSET      (ADSP_SRAM_INBOX_OFFSET \
+				     + ADSP_SRAM_INBOX_SIZE)
+
+#define ADSP_SRAM_EXCEPT_BASE       (ADSP_SRAM_DEBUG_BASE \
+				     + ADSP_SRAM_DEBUG_SIZE)
+#define ADSP_SRAM_EXCEPT_SIZE       0x800
+#define ADSP_SRAM_EXCEPT_OFFSET     (ADSP_SRAM_DEBUG_OFFSET \
+				     + ADSP_SRAM_DEBUG_SIZE)
+
+#define ADSP_SRAM_STREAM_BASE       (ADSP_SRAM_EXCEPT_BASE \
+				     + ADSP_SRAM_EXCEPT_SIZE)
+#define ADSP_SRAM_STREAM_SIZE       0x1000
+#define ADSP_SRAM_STREAM_OFFSET     (ADSP_SRAM_EXCEPT_OFFSET \
+				     + ADSP_SRAM_EXCEPT_SIZE)
+
+#define ADSP_SRAM_TRACE_BASE        (ADSP_SRAM_STREAM_BASE \
+				     + ADSP_SRAM_STREAM_SIZE)
+#define ADSP_SRAM_TRACE_SIZE        0x1000
+#define ADSP_SRAM_TRACE_OFFSET      (ADSP_SRAM_STREAM_OFFSET \
+				     + ADSP_SRAM_STREAM_SIZE)
+
+#define ADSP_IMX8_DSP_MAILBOX_SIZE  (ADSP_SRAM_INBOX_SIZE \
+				     + ADSP_SRAM_OUTBOX_SIZE \
+				     + ADSP_SRAM_DEBUG_SIZE \
+				     + ADSP_SRAM_EXCEPT_SIZE \
+				     + ADSP_SRAM_STREAM_SIZE \
+				     + ADSP_SRAM_TRACE_SIZE)
+#define ADSP_IMX8_IRAM_SIZE          0x8000
+#define ADSP_IMX8_DRAM_SIZE          0x8000
+
+#define ADSP_IMX8_DSP_MU_SIZE         0x10000
+#define XSHAL_MU13_SIDEB_BYPASS_PADDR 0x5D310000
+#define ADSP_IMX8_DSP_MUB_BASE         XSHAL_MU13_SIDEB_BYPASS_PADDR
+#define ADSP_IMX8_DSP_MUA_BASE		0x5D280000
+
+#define ADSP_MAILBOX_SIZE			0x1000
+
+/* TODO get from driver. */
+#define IMX8_PANIC_OFFSET(x)	(x)
```

Comment:
Panic reporting on IMX8 is unsupported at this time; I need to focus on this at some point. Just to add why we don't have an actual value here.